docs/dev/porting_legacy_tests.md

```diff
@@ -0,0 +1,88 @@
+```
+
+See also: 📄 [Documentation for the `fill` command.](../filling_tests/filling_tests_command_line.md)
+
+> If the tests can't currently be filled, please explain the issue (feel free to also [open a Discussion](https://github.com/ethereum/execution-spec-tests/discussions/new?category=general)).
+
+## Debugging tests
+
+By default, EVM logs are stored in the `logs` folder at the repository root. You can check the `output` folder to review transaction results. If needed, review a previous PR that ported tests (e.g., [the PR porting the `PUSH` opcode](https://github.com/ethereum/execution-spec-tests/pull/975)).
+
+## Test coverage
+
+It's crucial that ported tests maintain coverage parity with legacy tests. This ensures that no critical functions are left untested and prevents introducing bugs. A CI workflow automatically checks for coverage.
+
+If coverage action fails (See: 📄 [An example of a failing test coverage](https://github.com/ethereum/execution-spec-tests/actions/runs/13037332959/job/36370897481)), it's recommended to run the coverage action locally (see: 📄 [How to run GitHub actions locally](./test_actions_locally.md)), which should generate a `evmtest_coverage` directory:
+
+```console
+❯ tree evmtest_coverage  -L 2
+evmtest_coverage
+└── coverage
+    ├── BASE
+    ├── BASE_TESTS
+    ├── coverage_BASE.lcov
+    ├── coverage_PATCH.lcov
+    ├── DIFF
+    ├── difflog.txt
+    ├── PATCH
+    └── PATCH_TESTS
+```
+
+Here `BASE`is legacy tests, `PATCH` is the ported test, and `DIFF` is the coverage difference on EVMONE. Open `evmtest_coverage/coverage/DIFF/index.html` in browser:
+
+![Annotated coverage](../img/annotated-coverage.jpg)
+
+| Label |                                   Description                                   |
+| ----- | :-----------------------------------------------------------------------------: |
+| `LBC` |    **Lost base coverage:** Code that was tested before, but is untested now.    |
+| `UBC` |  **Uncovered baseline code:** Code that was untested before and untested now.   |
+| `GBC` | **Gained baseline coverage:** Code that was untested before, but is tested now. |
+| `CBC` |    **Covered baseline code:** Code that was tested before and is tested now.    |
+
+Follow the hyperlinks for lost base coverage (`LBC`) to address coverage gaps. Here is an example coverage loss:
+
+![Missing legacy coverage](../img/legacy-coverage-loss.png)
+
+> Lost line coverage from a coverage report. In this case, caused by a missing invocation of `CALLDATALOAD`.
+
+!!! note "Expected coverage loss"
+
+    EEST uses [pytest](https://docs.pytest.org/en/stable/), a Python testing library, for testing Ethereum specifications, while legacy tests relied on the EVM to handle much of the execution. This difference can lead to coverage gaps. EEST favors dynamic contract creation for each test vector, while legacy tests preferred a single static contract with multiple test vectors determined by transaction input data.
```

Comment:
as a note here that yul tests or some tests used calldataload that might no longer needed when designing a test with python. but we should always investigate if an opcode is not covered anymore to see if its ok